jupyterhub_moss/spawner.py

```diff
@@ -213,12 +224,23 @@ def options_from_form(self, formdata: Dict[str, List[str]]) -> Dict[str, str]:
                 raise RuntimeError("GPU(s) not available for this partition")
             options["gres"] = gpu_gres_template.format(options["ngpus"])
 
+        partition_environments = tuple(
+            self.partitions[partition]["jupyter_environments"].values()
+        )
         if "environment_path" not in options:
             # Set path to use from first environment for the current partition
-            default_venv = tuple(
-                self.partitions[partition]["jupyter_environments"].values()
-            )[0]
-            options["environment_path"] = default_venv["path"]
+            options["environment_path"] = partition_environments[0]["path"]
+
+        corresponding_default_env = find(
+            lambda env: env["path"] == options["environment_path"],
+            partition_environments,
+        )
+        # custom envs are always added to PATH, defaults ones only if add_to_path is True
```

Comment:
Hope this is clearer, else feel free to provide suggestion

Reply:
That's great :+1: